src/types.rs

```diff
@@ -30,6 +33,12 @@ pub enum Error {
     /// HTTP request failure
     #[error("HTTP request failure: {0}")]
     Http(#[from] hyper::Error),
+    /// HTTP request failure
+    #[error("HTTP request failure: {0}")]
+    HttpHttp(#[from] hyper::http::Error),
+    /// HTTP client request failure
+    #[error("HTTP request failure: {0}")]
+    HttpClient(#[from] hyper_util::client::legacy::Error),
```

Comment:
I don't want `hyper_util` to appear in the public API, let's do an `Other<Box<dyn StdError + Send + Sync + 'static>>`?